qtrade/questrade.py

```diff
@@ -513,3 +513,168 @@ def submit_order(self, acct_id: int, order_dict: Dict[str, Union[int, bool, str]
         )
 
         return response
+
+    def lookup_symbol_by_name(self, symbolPrefix: str) -> List:
+        """Lookup symbol metadata by name. Can return 0, 1, or multiple dictionaries.
+
+           See https://www.questrade.com/api/documentation/rest-operations/market-calls/symbols-search
+
+           This method will return a dictionary of the form
+
+           ``[
+               {
+                 "symbol": "BMO",
+                 "symbolId": 9292,
+                 "description": "BANK OF MONTREAL",
+                 "securityType": "Stock",
+                 "listingExchange": "NYSE",
+                 "isTradable": true,
+                 "isQuotable": true,
+                 "currency": "USD"
+               }
+             ]``
+
+            Parameters
+            ----------
+            symbolPrefix: str
+                Symbol search prefix
+
+            Returns
+            -------
+            list:
+                List of dictionaries containing symbol metadata.
+            """
+        payload = {
+            "prefix": symbolPrefix,
+        }
+        log.info("Searching for symbol prefix {0} ...".format(symbolPrefix))
+        response = self._send_message("get", "symbols/search", params=payload)
+        return response
+
+    def get_option_chain(self, symbolId: int) -> Dict:
```

Comment:
I am wondering, I designed most methods such that one could call the method using a ticker symbol instead of this internal Questrade symbol ID. Internally it essentially calls `ticker_information` to translate the ticker into the ID and then calls the actual API call you want to make. I think it would make sense to follow this design in this case as well. See, for example, the `get_quote`-method.

Reply:
Done